server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -322,22 +323,12 @@ public List<Segment> segments(boolean verbose) {
 
     @Override
     public void refresh(String source) throws EngineException {
-        maybeRefresh(source);
+        // Refresh on this engine should only ever happen in the reader after new segments arrive.
     }
```

Comment:
Should we throw an `UnsupportedOperationException` instead to detect any hidden invocation?

Reply:
Will give this a try and run tests to see where it breaks.  I'm a bit worried we'll have to add too much conditional logic where refresh is invoked from IndexShard & api, but it may not be too bad...

Reply:
@Bukhtawar Took a look at this, I would prefer we not have implementation specific logic in the shard and leave as a NoOp in the Engine implementation.  There are 6 invocations of the blocking refresh on the abstract Engine from IndexShard.  All of these methods are still invoked on the shard so we would need conditional logic to skip the call or significant refactoring at a higher level.

- Direct refresh - API - req still routed to replica.
- acquireLastIndexCommitAndRefresh - not invoked with nrt replicas
- postRecovery - still invoked through multiple recovery paths
- finalizeRecovery  - still invoked through multiple recovery paths
- resetEngineToGLobalCheckpoint - used to reset engine after failover/relocation.